packages/spruce-types/src/events/get-ui-enhancements.ts

```diff
@@ -0,0 +1,26 @@
+import { IHWCalendarEventDetailsItem, IHWAction } from '../generated/hw-gql'
+
+export interface IEventGetUIEnhancementsPayload {
+	/** The view that is being displayed. booking:create-appointment for example */
+	view: string
+
+	/** Other arbitrary payload properties defined by the specific view */
+	[key: string]: any
+}
+
+export interface IUIEnhancementSection {
+	/** The section id for this enhancement */
+	id: string
+
+	/** Event detail items to place in this section */
+	eventDetailsItems?: IHWCalendarEventDetailsItem[]
+
+	/** Actions that should be added to the context menu of this section */
+	actions?: IHWAction[]
```

Comment:
@gcpantazis the idea here is that these are actions that would get added to the context menu of each section - the `guest` or `services` section. The `calendarEventDetailsItems` could still have actions but it would be specific to the individual enhancement instead of the context menu.

Does that sound right to you? The alternative is that any action inside `calendarEventDetailsItems` would get pulled up into the context menu

Reply:
Hmm, if I'm understanding you correctly I'm not sure how we'd determine which actions go where, so you'd have to put them in **every** context menu? i.e. "Guest Notes" in this screen:

![Screen Shot 2019-10-30 at 4 16 59 PM](https://user-images.githubusercontent.com/1161192/67899702-db6f0980-fb30-11e9-9e0e-6196b395dee3.png)

I think better might be to treat these the same as eventDetailsItems; make a new key `contextMenuItems? : IHWButton[]` (which is what the context menu items are). Then I'd specifically request the sectionId like, `guest:context_menu_items` or some such.

Reply:
Resolved to keep `actions` after discussion